modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -2597,8 +2600,35 @@ private GridTuple3<Long, Long, Boolean> ttlAndExpireTime(IgniteCacheExpiryPolicy
      */
     private void drReplicate(GridDrType drType, @Nullable CacheObject val, GridCacheVersion ver, AffinityTopologyVersion topVer)
         throws IgniteCheckedException {
-        if (cctx.isDrEnabled() && drType != DR_NONE && !isInternal())
+        GridToStringBuilder.SensitiveDataLogging sensitiveDataLogging = S.getSensitiveDataLogging();
+
+        String keyValue = "nil";
+
+        if (log.isTraceEnabled()) {
+            if (sensitiveDataLogging == PLAIN) {
+                keyValue = keyValue(false).toString();
+            } else if (sensitiveDataLogging == HASH) {
+                keyValue = String.valueOf(IgniteUtils.hash(keyValue(false)));
+            }
+
+            if (!((GridNearCacheEntry) this).recordDhtVersion(ver)) {
+                logOnMoreRecent();
+            }
+        }
+
+        traceReplicate(ver, keyValue);
```

Comment:
1. Let's move all this logic under the `if (cctx.isDrEnabled() && drType != DR_NONE && !isInternal()) {` statement.
Because there could be non-DR caches, which shouldn't be logged.

2. Let's extract inner if statement into a static method.